exonum/src/runtime/rust/mod.rs

```diff
@@ -121,12 +137,15 @@ impl RustRuntime {
             .expect("Method called before Rust runtime is initialized")
     }
 
+    /// Adds a new service factory to the list of available artifacts.
     pub fn add_service_factory(&mut self, service_factory: Box<dyn ServiceFactory>) {
         let artifact = service_factory.artifact_id();
         trace!("Added available artifact {}", artifact);
         self.available_artifacts.insert(artifact, service_factory);
     }
 
+    /// Adds a new service factory to the list of available artifacts and returns
```

Comment:
Same as above + I'd change "returns itself" to "returns modified `RustRuntime` object".